go/mcap/range_index_heap.go

```diff
@@ -0,0 +1,41 @@
+package mcap
+
+type rangeIndex struct {
+	chunkIndex        *ChunkIndex
+	messageIndexEntry *MessageIndexEntry
+	reverse           bool
```

Comment:
This should be the same for all entries in the heap, right? Maybe we can omit it from the heap entries themselves?